jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ProjectAccountSettingsManager.kt

```diff
@@ -64,6 +75,21 @@ interface ProjectAccountSettingsManager {
      */
     fun recentlyUsedCredentials(): List<ToolkitCredentialsProvider>
 
+    /**
+     * Attempts to change the active credential provider.
+     *
+     * 1. Broadcasts a [AccountSettingsChangedNotifier] with [AccountSettingsEvent.isLoading] set to true
+     * 2. Kicks off an STS callerIdentity call to validate the credentials work in the background
+     * 3. If call succeeds, make the requested provider active, broadcast [AccountSettingsEvent.activeCredentialsChanged]
+     * 4. If call fails, null out the active provider, broadcast [AccountSettingsEvent.activeCredentialsChanged]
+     */
+    fun changeCredentialProvider(credentialsProvider: ToolkitCredentialsProvider?)
```

Comment:
The function comment seems like an implementation detail, which would belong on the `DefaultProjectAccountSettingsManager` version of `changeCredentialProvider`. Can it be on both?

Reply:
Callers of the code never seen the true method since it is hidden behind the interface in the dependency injection system.

Javadoc / kdoc is also inherited on overriden methods unless explicitly overridden.